### PR TITLE
Feature/1868 experiment language

### PR DESCRIPTION
**Pull request by timohuber**

### CODE REVIEW

File: pool/app/contact/contact.ml [hunk 1]
```diff
@@ -31,6 +31,7 @@ let find_full_cell_phone_verification_by_contact =
 
 let has_terms_accepted = Event.has_terms_accepted
 
+(* TODO: Can this be removed? *)
```
Comment (mabiede) [inline pool/app/contact/contact.ml:34]: ping

File: pool/app/message_template/message_template.ml [hunk 2]
```diff
@@ -80,6 +79,7 @@ let prepare_email language template sender email layout params =
   Message_utils.render_email_params params mail
 ;;
 
+(* TODO: Should this be using experiment language as well? *)
```
Comment (mabiede) [inline pool/app/message_template/message_template.ml:82]: ping

File: pool/app/message_template/message_utils.ml [hunk 3]
```diff
@@ -178,16 +178,35 @@ let combine_html language html_title =
...
-  let open Pool_common in
-  let open CCOption in
-  let default = Settings.default_language_of_list sys in
-  contact.Contact.language
-  >>= (fun contact_lang -> CCList.find_opt (Language.equal contact_lang) sys)
-  |> value ~default
-;;
-
-let search_by_language templates lang =
+module Old = struct
```
Comment (mabiede) [inline pool/app/message_template/message_utils.ml:181]: ping

File: pool/app/pool_common/locales/i18n_de.ml [hunk 4]
```diff
@@ -342,11 +346,10 @@ Wenn Sie die Erinnerungen jetzt manuell auslösen werden über den gewählten Na
      Testnachricht schicken können, um den API Key zu verifizieren. Die Nummer \
      muss im Format +41791234567 sein."
   | I18nText str -> str
-  | MissingMessageTemplates (label, languages) ->
-    Format.asprintf
-      "Das '%s' Template fehlt in den folgenden Sprachen: %s"
-      label
-      (languages |> CCString.concat ", ")
+  | MissingMessageTemplates ->
+    "Die folgenden Nachrichtenvorlagen sind nicht vorhanden. Die \
```
Comment (mabiede) [inline pool/app/pool_common/locales/i18n_de.ml:350]: welche folgenden? (german and english)

File: pool/test/message_template_test.ml [hunk 5]
```diff
@@ -109,6 +109,138 @@ let create_invitation language ?entity_uuid () =
...
+  let find_template = find_template_to_send experiment in
+  let%lwt res_de = find_template contact_de in
+  let%lwt res_en = find_template contact_en in
+  (* Always expect the template to be in experiment language *)
+  let () =
+    Alcotest.(check Test_utils.message_template "succeeds" template_de res_de)
+  in
+  let () =
+    Alcotest.(check Test_utils.message_template "succeeds" template_de res_en)
+  in
```
Comment (mabiede) [inline pool/test/message_template_test.ml:240]: Define a partially applied function called `check_experiment_message_template` with one input argument for the English and German contact. With this, the previous comment becomes obsolete.

File: pool/test/test_utils.ml [hunk 6]
```diff
@@ -115,15 +115,21 @@ module Model = struct
       }
   ;;
 
-  let create_contact ?id ?name ?(with_terms_accepted = true) () =
+  let create_contact
+    ?id
+    ?(language = Pool_common.Language.En)
```
Comment (mabiede) [inline pool/test/test_utils.ml:120]: Is this default value expected? Should be possible to directly pass the optional argument to the record as a contact shouldn't get a language by default.

File: pool/web/view/component/component_input.ml [hunk 7]
```diff
@@ -749,7 +733,14 @@ let multi_select
   in
   div
     ~a:[ a_class (Elements.group_class classnames orientation) ]
-    ([ label [ txt (Elements.input_label language group_field None required) ]
+    ([ label
+         [ txt
+             (Elements.input_label
+                language
```
Comment (mabiede) [inline pool/web/view/component/component_input.ml:739]: Somewhere, the label of the input element gets the other language then it's actually chosen by the contact (when EN is selected, the label is in German and vice versa).

File: pool/web/view/component/component_input.ml [hunk 8]
```diff
@@ -749,7 +733,14 @@ let multi_select
   in
   div
     ~a:[ a_class (Elements.group_class classnames orientation) ]
-    ([ label [ txt (Elements.input_label language group_field None required) ]
+    ([ label
```
Comment (mabiede) [inline pool/web/view/component/component_input.ml:736]: Add contact hint: add hint: some experiments choose to send invitations in a different language, disregarding your contact language

**Review by mabiede — CHANGES_REQUESTED** [review]

File: pool/app/pool_common/locales/i18n_de.ml (hunk 4 above)
Comment (timohuber) [reply pool/app/pool_common/locales/i18n_de.ml:350]: Diese: https://github.com/uzh/pool/pull/279/files#diff-5958c332d013c1af8cf8f84f61856eee77ff6129afedb5864a91bd23e62f3fa4R62

File: pool/web/view/component/component_input.ml (hunk 7 above)
Comment (timohuber) [reply pool/web/view/component/component_input.ml:739]: Fixed, refreshing the page when language is updated

File: pool/app/pool_common/locales/i18n_de.ml (hunk 4 above)
Comment (mabiede) [reply pool/app/pool_common/locales/i18n_de.ml:353]: a list called list^^

**Review by mabiede — APPROVED** [review]

**Pull request merged**